### ISSUE TITLE
setting properties with awful.spawn()

### ISSUE THREAD

**Issue by paoloap** [awesomeWM/awesome#1211]
Output of `awesome --version`:

awesome v3.5.9 (Mighty Ravendark)
 • Build: Mar 12 2016 01:11:40 for x86_64 by gcc version 5.3.0 (builduser@rw)
 • Compiled against Lua 5.3.2 (running with Lua 5.3)
 • D-Bus support: ✔

**How to reproduce the issue:**

I want to open a VLC instance with awful.spawn and automatically send it to a specified tag when typing a shortcut. To do it, I would like to set its "instance" property to a specified value while executing it.

So, into globalkeys section, I add the following:

    awful.key({ modkey, "Mod1" }, "b", function ()
      awful.util.spawn("vlc", { name = "tv" } )
    end),

(considering that, from documentation, the method is configured this way):
    spawn (cmd[, sn_rules=true[, callback]])
    sn_rules table or boolean A table of properties to be applied after startup; false to disable startup notifications. (default true)

**Actual result:**

After awesomeWM restart, I try to execute the shortcut but get the following error:
    /usr/share/awesome/lib/util.lua:80: bad argument #2 to 'spawn' (boolean expected, got table)

**Expected result:**

Why in documentation is stated that sn_rules can be set as a table, if in fact it accepts only a boolean value? Did I misunderstand something?

Thanks!

**Comment by paoloap** [comment]
Uh, I just found out that the error was caused by the fact that I was using awful.util.spawn instead of awful.spawn (the first one keeps only a boolean as first parameter). Nevertheless, seems that awful.spawn method isn't recognized in my awesome version... what version is needed to make it usable?

**Comment by psychon** [comment]
`awful.util.spawn`: This is for awesome 3.5. In git/master, this was moved to `awful.spawn`.
`awful.spawn`: This is for the version from git/master, but not available in 3.5.9.

Also, setting `instance` won't work and setting `name` does not do what you would like it to do. Next problem is that (I think) `vlc` does not have the needed startup-notification support so that awesome can make a connection between `I started a new process` and `a new window appeared`. Thus, I think I'm sorry, but this whole approach cannot work.

The best option would be to do something like the following:
```lua
local started_vlc = false

-- Then in your awful.rules.rules table, do something like
callback = function(c) if started_vlc then c:tags(tags[1][2]) end started_vlc = false end -- This assigns some tag

-- And in your key callback do:
function()
  started_vlc = true
  awful.util.spawn("vlc")
end
```
